### DISCUSSION TITLE
Can you explain `<PackageReference>` with `<IncludeAssets>` in a particular example?

### DISCUSSION THREAD

**Discussion by MartyIX** [dotnet/runtime#59934] [Q&A]
VS2022 Preview 4.1 added the following snippet to my `.csproj` file when I added `xunit.runner.visualstudio` using VS2022's "Manage NuGet packages for solution":

```xml
<PackageReference Include="xunit.runner.visualstudio" Version="2.4.3">
  <PrivateAssets>all</PrivateAssets>
  <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
</PackageReference>  
```

Why it was added in this form really? 

* I understand that `<PrivateAssets>all</PrivateAssets>` was added to avoid the `xunit.runner.visualstudio` to be "inherited" by child C# projects. Right?
* `<IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>` why these values exactly? What would be the difference if the default `all` value were used?

Is it equivalent to write:

```xml
<PackageReference Include="xunit.runner.visualstudio" Version="2.4.3" PrivateAssets="all"> <!-- NOTE THE ATTRIBUTE -->
  <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
</PackageReference>  
```
?

PS: Useful resource https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#controlling-dependency-assets

**Comment by huoyaoyuan** [comment]
> Is it equivalent to write:

I don't think msbuild syntax can exchange xml attribute and element like xaml.

**Reply by MartyIX** [reply]
I have already seen it elsewhere so I think it is an alternative syntax. Also MSVS autocomplete tells me that the attribute exists. So maybe it's new syntax or deprecated syntax ... I don't really know.
